include/pisa/command/cli.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <CLI/CLI.hpp>
+
+namespace pisa::cli {
+
+namespace options {
+
+    template <typename Settings>
+    auto threads(CLI::App &app, Settings &settings)
+    {
+        return app.add_option("-j,--threads", settings.threads, "Thread count");
+    }
+
+    template <typename Settings>
+    auto batch_size(CLI::App &app, Settings &settings)
+    {
+        return app.add_option("-b,--batch-size",
+                              settings.batch_size,
+                              "Number of documents to process at a time",
+                              true);
+    }
+
+} // namespace options
+} // namespace pisa::cli
+
+#define PISA_MAIN(NAME, FUNCTION)                                                        \
```

Comment:
I think we can avoid the macro here.
Let's use a function which takes 2 functions as parameters: `settings_name` and `function`.
Then we are supposed to call this function inside a `main`.